app/controllers/admin/invisiblizes_controller.rb

```diff
@@ -1,7 +1,8 @@
 module Admin
   class InvisiblizesController < ApplicationController
     def create
-      Developer.find(params[:developer_id]).invisiblize!
+      developer = Developer.find(params[:developer_id])
+      InvisibleDeveloper.new(developer).mark
```

Comment:
aren't you worried the class' name and method's name aren't very explicit about what they do ?
In effect it looks like it's mainly to notify the developer of the invisible character of the profile, why not reflect that in one or both of the names ?

Reply:
I'm not worried about conveying the _entire_ functionality of a service in the name. `InvisibleDeveloper` tells me the core of what it does - marks a developer as invisible. Sending the notification, in my opinion, is part of that flow and fits better in here then in an Active Record callback.

That said, what would you name this if you _had_ to include the concept of sending notifications, too?